pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeFormatPatternSpec.java

```diff
@@ -81,6 +82,23 @@ public DateTimeFormatter getDateTimeFormatter() {
     return _dateTimeFormatter;
   }
 
+  /**
+   * Validates the sdf pattern
+   */
+  public static void validateFormat(@Nonnull String sdfPatternWithTz) {
```

Comment:
(convention) Let's not use `Nonnull` annotation because it is very easy to mix it with `Nullable`. By default we treat all fields as non-null, and only annotate the ones that might be null